js/src/controls/LayersControl.js

```diff
@@ -51,13 +73,22 @@ var LeafletLayersControlView = LeafletControlView.extend({
                 }
                 return ov;
             }, {});
+
+            var sel = that.model.get('selected');
             var overlays = views.reduce(function (ov, view) {
                 if (!(view.model.get("base")))
                 {
-                    ov[view.model.get("name")] = view.obj;
+                    var name = view.model.get("name");
```

Comment:
Don't you need the same logic for baselayers as well?

Reply:
If a base layer as the same name as an overlay layer, it will be a problem.
Or we do two distinct attributes.

Reply:
Good idea, two distinct attributes makes a lot of sense. Something like `selected_layer` and `selected_overlays` ?

Reply:
Something like that, but i think we will have a problem.
When you create a `LayersControl`, all the Layer are displayed. So they are all in `selected`. 
You check / uncheck some checkboxes and u decide to add a new `Layer` on the map. I'm afraid that `selected` will be completly wrong.

I saw something strange too. The object `layer` in JS have not an attribute `name` with the name of the layer in the event.
I don't know why because they have the attribute in the class definition.

Reply:
> The object layer in JS have not an attribute name 

Ooh.. the layers in `LeafletJS` don't have a name. You only specify a name when you add a new layer to the control, isn't it?